pandas/plotting/_core.py

```diff
@@ -1582,7 +1582,7 @@ def pie(self, **kwargs):
             raise ValueError("pie requires either y column or 'subplots=True'")
         return self(kind="pie", **kwargs)
 
-    def scatter(self, x, y, s=None, c=None, **kwargs):
+    def scatter(self, x, y, size=None, s=None, color=None, c=None, **kwargs):
```

Comment:
this can be used as positional arguments and your change is not backwards compatible, best to use the order:

`self, x, y, s, c, size, color, **kwargs`

Reply:
actually I misread the purpose of this PR, shouldn't we be replicating matplotlibs arguments? Can you indicate why you want to make this change?

Reply:
> this can be used as positional arguments

ah yes, thanks - @mosc9575 please ignore part of my previous comment

let's keep `s` and `c` in the signature then, add `color` and `size` after them, and deprecate `s` and `c`?

Reply:
> Can you indicate why you want to make this change?

Just for consistency really, as in `bar` and `barh` it's `color`

Reply:
> actually I misread the purpose of this PR, shouldn't we be replicating matplotlibs arguments? Can you indicate why you want to make this change?

I opend this issue, because `c` was not intuitive for me. I don't have the matplotlib background, I normally work with bokeh, and they have the same functionality of `c` named `color`. Then I was searching the pandas documentation and found `color` used for `hbar` and `bar`. Therefor I thought this is not consitent and a misstake by this function. To be honest, I first understood the namen when I looked into matplotlib today. But it is still not "intuitive" to me.

Please let me know if you think this shouldn't changed at all.

Reply:
I was reading the matplotlib documentation for a whilw and the same problem can be found there. I will wait for a response to my issue [#21795](https://github.com/matplotlib/matplotlib/issues/21795) and change both in one step, if this is approved.

Reply:
i think we could also accept color size and and not deprecate

Reply:
Yes, agreed, I'd go with:

- keep signature as is
- if `c` is `None`, take `color` from `kwargs` (likewise for `s` and `size`)
- raise an error if someone passes both `color` and `c`, or `size` and `s` (and test this)

Conversely, for bar:
- keep signature as is
- if `color` is `None`, take `c` from `kwargs`
- raise an error if someone passes both `color` and `c` (and test this)

Reply:
Matplolib won't change and I think the sugessted changes by @MarcoGorelli are good. I try to edit my PR soon.